airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/record.py

```diff
@@ -112,13 +112,14 @@ def record_resolve_id(self, record: MutableMapping[str, Any]) -> MutableMapping[
         # while resolving the `id` in `record_resolve_id`,
         # we re-assign the original id like `"gid://shopify/Order/19435458986123"`,
         # into `admin_graphql_api_id` have the ability to identify the record oigin correctly in subsequent actions.
+        # IF NOT `id` field is provided by the query results, we should return composed record `as is`.
         id = record.get("id")
-        if isinstance(id, str):
+        if id and isinstance(id, str):
             record["admin_graphql_api_id"] = id
             # extracting the int(id) and reassign
             record["id"] = self.tools.resolve_str_id(id)
             return record
-        elif isinstance(id, int):
+        elif not id or isinstance(id, int):
             return record
```

Comment:
Is there any scenario where nothing(None) can be returned from record_resolve_id? Then we should update the return type to `option[MutableMapping[str, Any]]`, right?

I mean, the elif seems very specific about not ID or ID being int. I am not sure if ID could be "something else" requiring us not to return a record.

Reply:
Also, I see this is kind of how the code existed before, so it is probably fine to just leave it with the new ID check.

Reply:
Generally speaking there are 2 cases:
1) When we have the `id` field, as `str`, we need to resolve it to int, up to the schema.
2) When there is no `id` field or it's already an `int` - so we need to skip the resolution, and return `as is`.

We can have 3 cases instead, but I'd like to keep it 2. If you feel it's worth having them separated like:

```python
if id and isinstance(id, str):
   ...
   return record
elif id and isinstance(id. int):
   return record
else:
   return record
```

We can have it.

Reply:
If there are only two possible ways, then I would prefer:
```
        id = record.get("id")
        if id and isinstance(id, str):
            record["admin_graphql_api_id"] = id
            # extracting the int(id) and reassign
            record["id"] = self.tools.resolve_str_id(id)
        return record
```
You return the record anyway, so you won't have to duplicate that line.

Reply:
Corrected.